### PR TITLE
[BACKLOG-2540] - AMD Conversion - Splitting ccc_wrapper into multiple files

### PR DESCRIPTION
**Pull request by dcleao**
* `VizTypeRegistry` now supports async factories, returning Promises.
* Added "common-ui/vizapi/ccc/wrapper/charts/all.js" as a way to load/reference all existing chart implementations (important for bundling)
* Made the `vizTypeProvider` leaner by removing the `def.js` dependency (`def.js` is only incurred when you actually wanna show CCC vizs)
* Removed ccc wrapper's "IE event bubbling shield"
* A bit refactoring of ccc wrapper label configuration functions
* A bit formatting of ccc wrapper code

@pminutillo, @DFieldFL please review.

### CODE REVIEW

**Comment by pminutillo** [comment]
@krivera-pentaho if you get any time, could you take a look at this pull request? Another set of eyes would be appreciated. Thanks.

File: package-res/resources/web/vizapi/ccc/wrapper/charts/AbstractChart.js [hunk 1]
```diff
@@ -0,0 +1,1587 @@
...
+			},
+	
+	        _getDiscreteColorScaleCore: function(memberPalette) {
+	            var colorRoleId = this._discreteColorRole;
+	            if(colorRoleId) {
+	                // 1 - The CCC color role is not being explicitly set, so whatever goes to the series role is used by the color role
+	                // 2 - When a measure discrim is used and there is only one measure, the CCC dim name of the discriminator is
+	                //      not of the "series" group; so in this case, there's no discriminator in the key
+	
+	                var colorAxis = this.axisByRole[colorRoleId],
```
Comment (pminutillo) [inline package-res/resources/web/vizapi/ccc/wrapper/charts/AbstractChart.js]: Indentation

File: package-res/resources/web/vizapi/ccc/wrapper/charts/AbstractChart.js [hunk 2]
```diff
@@ -0,0 +1,1587 @@
...
+	                    if(mask) {
+	                        var dimName = gem.cccDimName,
+	                        	dim = dims[dimName] || (dims[dimName] = {});
+	                        dim.format = mask;
+	                    }
+	                });
+	            }
+	        },
+			
+			_configureLabels: function(options, vizOptions) {
```
Comment (pminutillo) [inline package-res/resources/web/vizapi/ccc/wrapper/charts/AbstractChart.js]: Indentation

File: package-res/resources/web/vizapi/ccc/wrapper/charts/AbstractNormalizedBarChart.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+], function(AbstractBarChart) {
+	
+	return AbstractBarChart.extend({
+		methods: {
+	        _options: {
+	            valuesNormalized: true,
+				stacked: true
+	        },
+			
+			_configure: function() {
```
Comment (pminutillo) [inline package-res/resources/web/vizapi/ccc/wrapper/charts/AbstractNormalizedBarChart.js]: Indentation

File: package-res/resources/web/vizapi/ccc/wrapper/charts/HeatGridChart.js [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+	        },
+			
+	        _processSelection: function(selectedDatums) {
+				
+				if(!selectedDatums.length) return selectedDatums;
+					
+				// Select Cols OR Rows
+				
+				var data = this._chart.data,
+					// columns are optional in the HG
```
Comment (pminutillo) [inline package-res/resources/web/vizapi/ccc/wrapper/charts/HeatGridChart.js]: Indentation

**Comment by dcleao** [comment]
Thanks Peter, I just notices I have this kind of problems a little everywhere, due to the new editor I'm trying out (Visual Studio Code)... I'll take a global look at indentation!

**Comment by dcleao** [comment]
Ok, just replaced the commit by an indented one... Thanks again Peter.

**Comment by pminutillo** [comment]
Thanks Duarte. This looks good to me.

File: package-res/resources/web/vizapi/VizController.js [hunk 5]
```diff
@@ -350,7 +350,7 @@ define([
         chartExists();
       } else {
         var chartDiv = this.visualPanelElement;
-        vizTypes.createByTypeAync(visualization, chartDiv, function(viz) {
+        vizTypes.createByTypeAync(visualization, chartDiv).then(function(viz) {
```
Comment (krivera-pentaho) [inline package-res/resources/web/vizapi/VizController.js]: "createByTypeAync". Is it supposed to be "createByTypeAsync" (Aync->Async)?

File: package-res/resources/web/vizapi/vizTypeRegistry.js [hunk 6]
```diff
@@ -95,19 +100,32 @@ define(["service!IVizTypeProvider"], function(vizTypeProviders) {
...
-        var viz = instFactory(vizType.id, arg);
-        onCreated(viz);
-      });
-    } else {
-      var className = vizType['class'];
-      var viz2;
-      eval('(viz2 = new ' + className + '(arg))');
-      onCreated(viz2);
-    }
+  VizTypeRegistry.prototype.createByTypeAync = function(vizType, arg) {
```
Comment (krivera-pentaho) [inline package-res/resources/web/vizapi/vizTypeRegistry.js]: Same comment here. I think this should be Async and not Aync

**Comment by krivera-pentaho** [comment]
Aside from my 2 comments, I am fine with the rest of the commit

File: package-res/resources/web/vizapi/VizController.js (hunk 5 above)
Comment (dcleao) [inline package-res/resources/web/vizapi/VizController.js]: Well spotted! This method is temporary and wasn't even unit-tested :-P I think it is only being used by the VizController, I'll correct that.

**Comment by DFieldFL** [comment]
Thumbs up!

**Pull request merged**